siaskynet/_upload.py

```diff
@@ -18,6 +18,67 @@ def default_upload_options():
     return obj
 
 
+def upload(self, upload_data, custom_opts=None):
+    """Uploads the given generic data and returns the skylink."""
+
+    response = self.upload_request(upload_data, custom_opts)
+    sia_url = utils.uri_skynet_prefix() + response.json()["skylink"]
+    response.close()
+    return sia_url
+
+
+def upload_request(self, upload_data, custom_opts=None):
+    """Uploads the given generic data and returns the response object."""
+
+    opts = default_upload_options()
+    opts.update(self.custom_opts)
+    if custom_opts is not None:
+        opts.update(custom_opts)
+
+    if len(upload_data) == 1:
+        fieldname = opts['portal_file_fieldname']
+        # this appears to be missing in go sdk; maybe the server ignores it?
+        filename = opts['custom_filename']
+    else:
+        if not opts['custom_dirname']:
+            raise ValueError("custom_dirname must be set when "
+                             "uploading multiple files")
+        fieldname = opts['portal_directory_file_fieldname']
+        filename = opts['custom_dirname']
+
+    params = {
+        'filename': filename,
+        # 'skykeyname': opts['skykey_name'],
+        # 'skykeyid': opts['skyket_id'],
+    }
+
+    ftuples = []
+    for filename, data in upload_data.items():
+        ftuples.append((fieldname,
+                        (filename, data)))
+
+    if len(upload_data) == 1:
+        data = ftuples[0][1][1]
+        if hasattr(data, '__iter__') and (
+                not isinstance(data, bytes) and
+                not isinstance(data, str) and
+                not hasattr(data, 'read')):
+            # an iterator for chunked uploading
```

Comment:
This looks good but it should be documented so people know this functionality is there

Reply:
is the documentation I added to `upload_file_request` sufficient?